packages/block/src/block.ts

```diff
@@ -223,9 +229,20 @@ export class Block {
   validateTransactions(stringError: true): string[]
   validateTransactions(stringError = false) {
     const errors: string[] = []
-
-    this.transactions.forEach(function (tx, i) {
+    const self = this
+    this.transactions.forEach((tx, i) => {
       const errs = <string[]>tx.validate(true)
+      if (self._common.isActivatedEIP(1559)) {
+        if (tx.transactionType === 2) {
+          if ((<FeeMarketEIP1559Transaction>tx).maxFeePerGas.lt(self.header.baseFeePerGas!)) {
+            errs.push('tx unable to pay base fee (EIP-1559 tx)')
+          }
+        } else {
+          if ((<Transaction>tx).gasPrice.lt(self.header.baseFeePerGas!)) {
+            errs.push('tx unable to pay base fee (non EIP-1559 tx)')
```

Comment:
removing `self` and using `this` now that the function was updated to arrow syntax

```suggestion
    this.transactions.forEach((tx, i) => {
      const errs = <string[]>tx.validate(true)
      if (this._common.isActivatedEIP(1559)) {
        if (tx.transactionType === 2) {
          if ((<FeeMarketEIP1559Transaction>tx).maxFeePerGas.lt(this.header.baseFeePerGas!)) {
            errs.push('tx unable to pay base fee (EIP-1559 tx)')
          }
        } else {
          if ((<Transaction>tx).gasPrice.lt(this.header.baseFeePerGas!)) {
            errs.push('tx unable to pay base fee (non EIP-1559 tx)')
```

Reply:
I think Jochem decided somewhat explicitly against that solution since it would produce a lot of overhead with again the need for some factory class and all that stuff similarly to what has been done for the tx library, I would agree that it's not worth the effort right now for just a single field. Would be very much desirable though nonetheless here to come to a cleaner solution.